modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -399,22 +400,22 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
                     return nodes.get(0);
                 })
                 .thenCompose(sqlNode -> {
-                    final boolean rwOp = dataModificationOp(sqlNode);
-                    final HybridTimestamp txTime = outerTx != null ? outerTx.readTimestamp() : rwOp ? null : clock.now();
+                    boolean rwOp = dataModificationOp(sqlNode);
+                    InternalTransaction outerTx0 =
```

Comment:
it's not 'outer' transaction, let's use just `tx` or smth like this

Reply:
It looks confusing - we have two internal transactions, here and the second one below. Can we have just one?

Reply:
Done, thanks.